src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,131 @@
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const todos: Todo[] = useAppSelector(state => state.todos);
+  const data = useAppSelector(state => state.filter);
+  const currentTodo = useAppSelector(state => state.currentTodo);
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">8</td>
+  useEffect(() => {
+    getTodos()
+      .then((tds) => {
+        dispatch(todosActions.getTodos(tds));
+      });
+  }, [dispatch]);
 
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
+  const normalizedQuery = data.query.toLowerCase().trim();
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">quo adipisci enim quam ut ab</p>
-            </td>
+  let filteredTodos = todos.filter(todo => (
+    todo.title.toLowerCase().trim().includes(normalizedQuery)
+  ));
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+  const toggleTodos = () => {
+    switch (data.status) {
+      case Select.all:
+        return filteredTodos;
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">9</td>
-            <td className="is-vcentered"> </td>
+      case Select.active:
+        filteredTodos = filteredTodos.filter(todo => !todo.completed);
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">molestiae perspiciatis ipsa</p>
-            </td>
+        return filteredTodos;
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+      case Select.completed:
+        filteredTodos = filteredTodos.filter(todo => todo.completed);
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">10</td>
+        return filteredTodos;
 
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
+      default:
+        return filteredTodos;
+    }
+  };
```

Comment:
It should be one function responsible for filtering, move to outside of the component.